src/components/extendedFilter/extendedFilter.tsx

```diff
@@ -0,0 +1,93 @@
+import React, { useState, useEffect } from 'react';
+import { IconButton } from '@mui/material';
+import { KeyboardArrowLeft, KeyboardArrowRight } from '@mui/icons-material';
+import './extendedFilter.scss';
+import { Course } from '../../types/components/componentType';
+
+const ExtendedFilter: React.FC = () => {
+  const [courses, setCourses] = useState<Course[]>([]);
+  const [selectedCategory, setSelectedCategory] = useState<string | null>(null);
+  const [startIndex, setStartIndex] = useState(0);
+
+  useEffect(() => {
+    // replace with a real api
+    const fetchListOfCategory = async () => {
+      try {
+        const temporaryData: Course[] = [
```

Comment:
Why do you use here async function without **await** construction? Or you will write **await**, when we will wait for **getting data** from backend?

Reply:
Yes, I'll correct this function, when we get backend